tests/src/test/scala/tests/commands/ExportCommandSuite.scala

```diff
@@ -193,4 +193,38 @@ class ExportCommandSuite extends tests.BaseSuite {
                     |@maven//:org.codehaus.mojo_animal-sniffer-annotations_1.14
                     |""".stripMargin
   )
+
+  // org.apiguardian:apiguardian-api has no dependencies
+  // we check it's included in the dependencies of libthrift.
+  checkDeps(
+    "target dependency are transitively included",
+    s"""|  - dependency: org.apache.thrift:libthrift:0.10.0
+        |    dependencies:
+        |      - mytargets:guava
+        |  - dependency: com.google.guava:guava:25.1-jre
+        |    dependencies:
+        |      - mytargets:apiguardian
+        |    targets:
+        |      - mytargets:guava
```

Comment:
Could you add exclusion of org.checkerframework_checker-qual_2.0.0 here please? If we expand the transitive during resolution, I think we'd end up not respecting the exclude sets.

Reply:
I added a test with the exclusion you mentioned.
If you meant to add the exclusion on the definition of `libthrift` (like below), then `checker-qual` would still be included. That's because there's no dependency "at the maven/ivy level" between libthrift and checker-qual.

This is the behavior that we observe with Pants today.

```scala
s"""|  - dependency: org.apache.thrift:libthrift:0.10.0
    |    dependencies:
    |      - mytargets:guava
    |    exclusions:
    |      - organization: org.checkerframework
    |        name: checker-qual
    |  - dependency: com.google.guava:guava:25.1-jre
    |    dependencies:
    |      - mytargets:apiguardian
    |..."""
```